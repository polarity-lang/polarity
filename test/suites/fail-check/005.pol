data Nat { Z, S(n: Nat) }

data Eq(a: Type, x y: a) {
    Refl(a: Type, x: a): Eq(a, x, x)
}

#[opaque]
let opaqueTwo: Nat {S(S(Z))}

-- | This proof typechecks
let p2: Eq(Nat, S(S(Z)), opaqueTwo) {Refl(Nat, S(S(Z)))}