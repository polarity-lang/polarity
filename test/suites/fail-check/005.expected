T-002

  × The following terms are not equal:
  │   1: Eq(Nat, S(S(Z)), S(S(Z)))
  │   2: Eq(Nat, S(S(Z)), opaqueTwo)
  │ 
   ╭─[005.pol:3:1]
 3 │ data Eq(a: Type, x y: a) {
 4 │     Refl(a: Type, x: a): Eq(a, x, x)
   ·                          ─────┬─────
   ·                               ╰── Source of (1)
 5 │ }
   ╰────
    ╭─[005.pol:10:1]
 10 │ -- | This proof typechecks
 11 │ let p2: Eq(Nat, S(S(Z)), opaqueTwo) {Refl(Nat, S(S(Z)))}
    ·         ─────────────┬─────────────  ─────────┬────────
    ·                      │                        ╰── While elaborating
    ·                      ╰── Source of (2)
    ╰────
